### DISCUSSION TITLE
2.0.4 (Testing needed)

### DISCUSSION THREAD

**Discussion by ransome1** [ransome1/sleek#625] [Show and tell]
While crafting version 2.0.4, I've streamlined and polished a significant amount of code and logic for a more stable application. However, there's a chance that some issues, whether minor or major, might have slipped through testing. Your help in testing this release is crucial. Please remember to back up your data before diving in.

https://github.com/ransome1/sleek/releases/tag/v2.0.4-rc.6

**Comment by qalisto** [comment]
My fundamental question is, for a Win11/N95-1.7ghz platform, is it to be expected that version 2.x is not going to be as "snappy" as v.1.x?
There are improvements in this Build, but, would like to manage expectations.
The UI/UX clearly has more utility - no doubt. #

**Reply by ransome1** [reply]
@qalisto have you tried turning off the animations as described here https://github.com/ransome1/sleek/issues/615#issuecomment-1874163632?

The rewrite incorporates React as a modern front-end framework and it is without a doubt, that it adds some weight, which was not there in the v1.3.x. But I don't think it should be one, that really should make a difference. 

There is also a chance, that the lack of "snappyness" you're perceiving is due to my deliberate choice of sticking to React's default animations. Which can now be turned off, if it helps. 

On my hardware (3 years old, low end laptop) sleek runs fast. I cannot see any spikes in CPU usage or memory consumption. It is hard to improve something I cannot reproduce.

**Reply by qalisto** [reply]
@ransome1 
(1)  **Disabled** animations via config.json
(2)  Am now testing rc. 6
Initial results are that my 41k txt file of ToDo's which employ avg 1.5 (+)Projects or (@)Tasks consumes:
      Mem Usage= ~280mb
      CPU= ~33%
 (3) New Task input still "laggy" - but not "tear out own hair" levels.
 FYI: I get the Windows "Blue Screen of Suspicion" as has been usual upon exe, bypassing via clicks to "Run Anyway".
 
(?) Question:  I have simply been downloading the latest Win .exe and Running *that* within context of all other Dependent Sleek files from an Original "Setup" install from the "1.3x" series.  IS *that* a _mistake_ that might be contributing to sluggishness?  Should I have reinstalled the _whole enchilada_ and even fully-uninstalled v. 1 before-hand?  Or is this not relevant to v.2 robustness (and in future are simple drop-in replacements of the *.exe an OK practice?)  ###

**Reply by ransome1** [reply]
I have limited experience with Windows in general and may not be the most reliable source for advice on the distinctions between various Windows releases. 

What I do know is that there seems to be a notable difference between the .exe portable release and a "regular" release, particularly during the application startup. The .exe release appears to be considerably slower due to the process of mounting during startup before the application itself is executed. While I can't provide specific details, this information might explain the observed performance variation.

If you encounter performance issues, you might consider downloading the .zip release, which includes the sleek.exe application. This executable is designed to be directly executed without the additional mounting step. Like this you will not have to "overwrite" the currently installed 1.3.x release. It will just run parallel. I'm curious if this changes anything on you side.

**Reply by ransome1** [reply]
>  Mem Usage= ~280mb

That should be alright, since it is still an Electron app.

>  CPU= ~33%

That is also the case on my system, but only when I'm wildly clicking around within the app and only accumulated from the several processes, which are running.

<img width="394" alt="Screenshot 2024-01-04 at 11 39 21 AM" src="https://github.com/ransome1/sleek/assets/11188741/c5795db8-e0f6-4695-8344-ad0ae995a153">
<img width="534" alt="Screenshot 2024-01-04 at 11 39 31 AM" src="https://github.com/ransome1/sleek/assets/11188741/b223c3b4-f815-45d9-b52d-2f89afdd976c">

**Reply by ransome1** [reply]
>  (2) Am now testing rc. 6

The `disableAnimations` setting is only available in the final version of `2.0.4`.

**Reply by ransome1** [reply]
For the sake of proper documentation, I think this discussion should be done in the corresponding issue thread: https://github.com/ransome1/sleek/issues/615

**Comment by qalisto** [comment]
> > (2) Am now testing rc. 6
> 
> The `disableAnimations` setting is only available in the final version of `2.0.4`.

Am a little confused, per att'd I made "True" the disablement which is clearly available in my config.json using the rc. 6  **.exe**

Unless execution of sleek.exe itself modifies the legacy/prior creation of config.sys (which absolutely came from a previous "setup" installation), then the "disable animations" setting came from a prior installation of either 1.x or maybe my first 2.x installation.

(If I should be testing the config modif in another version, pls advise - and if you like we can shift this back to the correct topical track [sorry 'bout that] .)

BTW, really like the new filtering/showing of all active due dates.  I observed that when they (and other attributes) are Expanded fully, it seems to eat more Resources.

Thx!  -Q

**Reply by ransome1** [reply]
> > > (2) Am now testing rc. 6
> > 
> > 
> > The `disableAnimations` setting is only available in the final version of `2.0.4`.
> 
> Am a little confused, per att'd I made "True" the disablement which is clearly available in my config.json using the rc. 6 **.exe**
> 
> Unless execution of sleek.exe itself modifies the legacy/prior creation of config.sys (which absolutely came from a previous "setup" installation), then the "disable animations" setting came from a prior installation of either 1.x or maybe my first 2.x installation.
> 
> (If I should be testing the config modif in another version, pls advise - and if you like we can shift this back to the correct topical track [sorry 'bout that] .)
> 
> BTW, really like the new filtering/showing of all active due dates. I observed that when they (and other attributes) are Expanded fully, it seems to eat more Resources.
> 
> Thx! -Q

I am not sure, if I understand what you are saying. Is it correct you are confused to find the `disableAnimations` setting?

If you downloaded `2.0.4` and started it, it will start a migration process of the `config.json`. With the update to `2.0.4` the new setting is added to the configuration file. This will have no effect on any old installations you might be using in parallel, since those version don't know what to do with this setting. It has nothing to do with `1.3.x`.

Since I don't know what is going on on your PC, the easiest thing might be to simply remove sleek and all configuration files. And then download the just released `2.0.5` and change the `config.json`, in order to disable the animations.
